store/repository.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	"github.com/odpf/siren/store/model"
+	"github.com/odpf/siren/store/postgres"
+	"gorm.io/gorm"
+)
+
+type NamespaceRepository interface {
+	Migrate() error
```

Comment:
@whoAbhishekSah Let's remove migrate from repositories and we can have a migrate method that takes care of overall database migration.

Reply:
I feel repository being the single interface for all DB-related actions, it should have the Migrate method, so that caller can use it. Otherwise, the caller will have to know the DB details.

Reply:
Migrate is a bootstrap process which is something services will not use. Check out Guardian ref. Only database implementation should know about migration.

Reply:
I'll fix this one while refactoring each package one by one.